Update permalink panel text to be 'View [postType]' instead of 'Preview'

brentswisher opened the pull request:
## Description
Fixes #14206 by replacing the text "Preview" in the permalink panel of the document settings with "View X" where X is the post type being edited. Preview can cause some confusion as there is also a "Preview" button at the top of the settings that does something different (generates a preview) where this link is just the currently live page.

## How has this been tested?
Open a Post, see it now says "View Post"
![Post](https://user-images.githubusercontent.com/6653970/59109873-ddc79180-890b-11e9-917e-378fdc030c76.png)

Open a Page, see it now says "View Page"
![Page](https://user-images.githubusercontent.com/6653970/59109886-e750f980-890b-11e9-9215-975a310b76d8.png)

Enable Gutenberg's custom post type plugin, add a PublicQueryPublic custom post, see that it says "View PublicQueryPublic"
![CustomPost](https://user-images.githubusercontent.com/6653970/59109897-ec15ad80-890b-11e9-80b0-18a1ea299291.png)

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

brentswisher committed:
Update permalink panel text to the view_item label from get_post_type_labels instead of 'Preview'

swissspidy commented on packages/edit-post/src/components/sidebar/post-link/index.js:
```diff
@@ -95,7 +96,10 @@ function PostLink( {
-				{ __( 'Preview' ) }
+				{ sprintf(
+					/* translators: %s: post type singular name */
+					__( 'View %s' ), postLabel
```
This should use the proper `view_item` label for this post type as generated by `get_post_type_labels` in WordPress. Anything else does not translate well.

swissspidy commented on packages/edit-post/src/components/sidebar/post-link/index.js:
```diff
@@ -148,6 +152,7 @@ export default compose( [
 			postTitle: getEditedPostAttribute( 'title' ),
 			postSlug: getEditedPostAttribute( 'slug' ),
 			postID: id,
+			postLabel: get( postType, [ 'labels', 'singular_name' ] ),
```
This should use the `view_item` label.

aduth commented:
Related Slack discussion concerning translatability ([link requires registration](https://make.wordpress.org/chat/)):

https://wordpress.slack.com/archives/C02QB2JS7/p1558731384027700

brentswisher commented:
Thank you for the speedy feedback @swissspidy and @aduth! I updated to use "view_item" and added a fallback to __( 'View Post' ) 

Also, I had missed that slack conversation @mapk , so sorry if I stepped on your toes here

aduth commented:
> so sorry if I stepped on your toes here

No toes were stepped on! 😃

mapk commented:
> Also, I had missed that slack conversation @mapk , so sorry if I stepped on your toes here

No worries. Thanks for picking this up! 😄 

I tested this and it works great and looks good in the UI. Let's merge this if the requests from @aduth are all good.

aduth reviewed (approved):
Looks good 👍
